BibBuddy/app/src/main/java/de/bibbuddy/BookOnlineFragment.java

```diff
@@ -77,18 +76,17 @@ private Bundle createBookBundle() {
     return bundle;
   }
 
-  /**
-   * Closes the BookOnlineFragment.
-   */
-  public void closeFragment() {
-    BookFragment fragment = new BookFragment();
-    getActivity().getSupportFragmentManager().beginTransaction()
-        .replace(R.id.fragment_container_view, fragment)
-        .setReorderingAllowed(true)
-        .addToBackStack(null)
-        .commit();
+  private void closeFragmentAfterAdded() {
+    Fragment fragment =
+        getActivity().getSupportFragmentManager().findFragmentByTag(LibraryKeys.FRAGMENT_BOOK);
```

Comment:
```suggestion
        requireActivity().getSupportFragmentManager().findFragmentByTag(LibraryKeys.FRAGMENT_BOOK);
```
getActivity() or getContext(), etc. may produce NullPointerExceptions - to solve this warning:
Replace all getActivity() or getContext() - cases with requireActivity() or requireContext() - you should do that for all classes you changed at least.

Reply:
✓